src/main.rs

```diff
@@ -350,6 +339,37 @@ struct CameraInterpolation {
     target_time: Instant,
 }
 
+impl CameraInterpolation {
+    fn new(camera: &Camera, scene_geometries: &[geometry::Geometry], time: Instant) -> Self {
+        let (source_origin, source_radius) = camera.visible_sphere();
+        let (target_origin, target_radius) = geometry::compute_bounding_sphere(&scene_geometries);
+
+        CameraInterpolation {
+            source_origin,
+            source_radius,
+            target_origin,
+            target_radius,
+            target_time: time + CAMERA_INTERPOLATION_DURATION,
+        }
+    }
+
+    fn duration_left(&self, time: Instant) -> f32 {
```

Comment:
Maybe have just a single function instead of these 3. Usage:

```rust
let (sphere_origin, sphere_radius) = interp.update(time);
camera.zoom_to_fit_visible_sphere(sphere_origin, sphere_radius);
```

Reply:
Done. It also takes easing as parameter now. Later on we could make this function generic and pass some `Easing` trait if we ever need more different types / configurations.